api/src/api/users.ts

```diff
@@ -1,20 +1,135 @@
 import express, { Request, Response } from 'express';
+import { ObjectId } from 'mongodb';
 import { errorWrap } from '../middleware';
 
 import User from '../models/user';
 
 const router = express.Router();
 
+/**
+ * Validates an ID to whether or not it is a valid MongoDB ID or not
+ * @param id Potential User ID to validate
+ */
+const isValidMongoId = (id: string): boolean => ObjectId.isValid(id);
+
+// gets all users
 router.get(
   '/',
   errorWrap(async (req: Request, res: Response) => {
-    const user = await User.findOne();
+    const user = await User.find({});
     res.status(200).json({
-      message: `Successfully retrieved ${user.name} users.`,
+      message: `Successfully retrieved all users.`,
       success: true,
       result: user,
     });
   }),
 );
 
+//gets user by id
+router.get(
+  '/:userId',
+  errorWrap(async (req: Request, res: Response) => {
+    if (!isValidMongoId(req.params.userId)) {
+      res.status(400).json({
+        success: false,
+        message: 'Bad ID format',
+      });
+      return;
+    }
+
+    const user = await User.findById(req.params.userId);
+    if (!user) {
+      res.status(404).json({
+        success: false,
+        message: 'User not found with id',
+      });
+    } else {
+      res.status(200).json({
+        success: true,
+        result: user,
+        message: `Successfully retrieved user`,
+      });
+    }
+  }),
+);
+
+// Create a new user
+router.post(
+  '/new',
```

Comment:
I would avoid defining a separate endpoint for this - you can just post to /users since all your users live here